docs/architecture/grpc-for-wcf-developers/call-credentials.md

```diff
@@ -1,11 +1,143 @@
+                {
+                    ValidateAudience = false,
+                    ValidateIssuer = false,
+                    ValidateActor = false,
+                    ValidateLifetime = true,
+                    IssuerSigningKey = signingKey
+                };
+        });
+
+}
+```
+
+The `IssuerSigningKey` property requires an implementation of `Microsoft.IdentityModels.Tokens.SecurityKey` with the cryptographic data necessary to validate the signed tokens. This token should be stored securely in a *secrets server* like Azure KeyVault.
+
+Next add the Authorization service, which controls access to the system.
+
+```csharp
+    services.AddAuthorization(options =>
+    {
+        options.AddPolicy(JwtBearerDefaults.AuthenticationScheme, policy =>
+        {
+            policy.AddAuthenticationSchemes(JwtBearerDefaults.AuthenticationScheme);
+            policy.RequireClaim(ClaimTypes.Name);
+        });
+    });
+
+```
+
+> [!TIP]
+> Authentication and Authorization are two separate steps. Authentication is used to determine the user's identity. Authorization decides whether that user is allowed to access various parts of the system.
+
+Now add the Authentication and Authorization middleware to the ASP.NET Core pipeline in the `Configure` method.
+
+```csharp
+public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+{
+    if (env.IsDevelopment())
+    {
+        app.UseDeveloperExceptionPage();
+    }
+
+    app.UseRouting();
+
+    // Authenticate, then Authorize
+    app.UseAuthentication();
+    app.UseAuthorization();
+
+    app.UseEndpoints(endpoints =>
+    {
+        endpoints.MapGrpcService<PortfolioService>();
+    });
+}
+```
+
+Finally, apply the `[Authorize]` attribute to any services or methods to be secured, and use the `User` property from the underlying `HttpContext` to verify permissions.
+
+```csharp
+[Authorize]
+public override async Task<GetResponse> Get(GetRequest request, ServerCallContext context)
+{
+    if (!TryValidateUser(request.TraderId, context.GetHttpContext().User))
+    {
+        throw new RpcException(new Status(StatusCode.PermissionDenied, "Denied."));
+    }
+
+    var portfolio = await _repository.GetAsync(traderId, request.PortfolioId);
+
+    return new GetResponse
+    {
+        Portfolio = Portfolio.FromRepositoryModel(portfolio)
+    };
+}
+```
+
+## Providing call credentials in the client application
+
+Once you've obtained a JWT token from an identity server, you can use it to authenticate gRPC calls from the client by adding it as a metadata header on the call, as follows:
+
+```csharp
+public async Task ShowPortfolioAsync(int portfolioId)
+{
+    var headers = new Grpc.Core.Metadata
+    {
+        { "Authorization", $"Bearer {_userToken}" }
+    };
+    var request = new GetRequest
+    {
+        TraderId = _userId,
+        PortfolioId = portfolioId
+    };
+    var response = await _portfoliosClient.GetAsync(request, headers);
+
+    // Display portfolio
+}
+```
```

Comment:
IMO add a tip that Grpc.Core.CallCredentials can be configured on a channel to centralize setting authentication metadata on calls.

Reply:
I see you show CallCredentials in other places. I still think a tip would be useful here for devs who don't read from beginning to end.